File: src/php/core/src/Style/MetaLicense.php
<?php

declare(strict_types=1);

namespace DiceBear\Style;

/**
 * Read-only view over the `meta.license` block of a style definition.
 */
class MetaLicense
{
    /**
     * @param array<string, mixed> $data
     */
    public function __construct(private readonly array $data) {}

    /**
     * Returns the license name (e.g. `"CC BY 4.0"`), or `null` when not set.
     */
    public function name(): ?string
    {
        return $this->data['name'] ?? null;
    }

    /**
     * Returns the license URL, or `null` when not set.
     */
    public function url(): ?string
    {
        return $this->data['url'] ?? null;
    }

    /**
     * Returns the full license text, or `null` when not set.
     */
    public function text(): ?string
    {
        return $this->data['text'] ?? null;
    }
}

File: src/rust/core/src/style/meta_license.rs
//! Read-only view over a style definition's `meta.license` block.

use serde::Deserialize;

#[derive(Deserialize, Default)]
pub(crate) struct MetaLicense {
    name: Option<String>,
    url: Option<String>,
    text: Option<String>,
}

impl MetaLicense {
    pub(crate) fn name(&self) -> Option<&str> {
        self.name.as_deref()
    }

    pub(crate) fn url(&self) -> Option<&str> {
        self.url.as_deref()
    }

    #[allow(dead_code)]
    pub(crate) fn text(&self) -> Option<&str> {
        self.text.as_deref()
    }
}
